src/RazorSdk/SourceGenerators/IncrementalValueProviderExtensions.cs

```diff
@@ -8,8 +8,55 @@
 
 namespace Microsoft.NET.Sdk.Razor.SourceGenerators
 {
+    /// <summary>
+    /// A highly specialized comparer that allows for treating an event source as cached if the razor options are set to suppress generation.
+    /// </summary>
+    /// <remarks>
+    /// This should not be used outside of <see cref="IncrementalValuesProviderExtensions.AsCachedIfSuppressed{T}(IncrementalValueProvider{T}, IncrementalValueProvider{RazorSourceGenerationOptions})"/>
```

Comment:
>IncrementalValuesProviderExtensions [](http://example.com/codeflow?start=54&length=35)

Consider making `RazorSourceGeneratorComparer<T>` a private nested class in `IncrementalValuesProviderExtensions`, to reduce the chance this will be used elsewhere.